### ISSUE TITLE
Generated Xcode project with prebuilt_cxx_library can't find library headers

### ISSUE THREAD

**Issue by ProfessaA** [facebook/buck#1098]
I have a simple buck project for a static iOS library and some tests that use `OCMock` (https://github.com/ProfessaA/BuckWithPrebuiltStaticLib).  

The way I have this setup is with OCMock's headers and fat static library in a folder being built using the `prebuilt_cxx_library` build rule.  The tests for the static library depend on the OCMock target, and import `<OCMock/OCMock.h>`.  Tests run fine from the command line, but when I try to run tests on the project generated with buck, I get an error that `<OCMock/OCMock.h>` cannot be found.  

Is there a way to get the generated Xcode project to know about the library headers so that I can edit my code in Xcode? Is there some other way more compatible with buck for including third party libs?

**Comment by alevinesc** [comment]
@ryu2 or @Coneko are there plans to support `prebuilt_cxx_library` in generated xcode projects?  If not, do you have suggestions for how to integrate third party libraries (like OCMock) into buck projects?

**Comment by ryu2** [comment]
It's on our radar for this half.

Internally, we pretty much build all third-party dependencies (such as OCMock) from source if possible.  For closed-source libs, a workaround is:

1. Declare a dummy library with the headers so you can include them as needed:
```
cxx_library(
  name = 'foobar',
  srcs = ['dummy.c'],
  exported_headers = subdir_glob([
    ('src', 'foobar_public/foobar.h')
  ]),
  visibility = ['PUBLIC'],
)
```

2. In your `apple_binary`, include the real library so that it links:
```
libraries = [
    'lib/libfoobar.a',
  ],
```
where you've checked in the actual `libfoobar.a` somewhere in your source tree.

**Comment by alevinesc** [comment]
Thanks!! The missing piece was the `libraries` field which doesn't seem to be documented.

There was one more thing I had to do that feels a little hacky.  When I tried to reference the third party lib from my `apple_library` target, I initially got a "refers to a parent directory" error and had to link the library to be in a child directory.

For building third-party dependencies from source, a difficulty I ran into was that a lot of projects have a Prefix header.  I tried to use `-include` in `compiler_flags` with the relative path of the .pch file, but the generated project didn't work.  Is there a better way to replicate Xcode's "Prefix Header" setting with buck?
